File: Data Structures and Algorithms/283-MoveZeroes/283-MoveZeroes.cpp
// Last updated: 6/4/2026, 6:35:48 PM
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int l = 0;
        for( int r=0; r < nums.size(); r++)
        {
            if (nums[r] != 0)
            {
                int temp = nums[r];
                nums[r] = nums[l];
                nums[l] = temp;
                l++;
            }
        }
    }
};

File: Data Structures and Algorithms/283-MoveZeroes/283-MoveZeroes.py
# Last updated: 6/4/2026, 6:35:39 PM
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        left = 0
        for right in range(0,len(nums)):
            if nums[right] != 0:
                temp = nums[right]
                nums[right] = nums[left]
                nums[left] = temp
                left+=1
        